src/include/gurt/debug_setup.h

```diff
@@ -0,0 +1,249 @@
+	ACTION(DB_ALL,	 all,   all,   0, arg)	\
+	/** Generic debug stream */		\
+	ACTION(DB_ANY,	 any,	any,   0, arg)	\
+	/** Extremely verbose debug stream */	\
+	ACTION(DB_TRACE, trace, trace, 0, arg)	\
+	/** Memory operations */		\
+	ACTION(DB_MEM,	 mem,	mem,   0, arg)	\
+	/** Network operations */		\
+	ACTION(DB_NET,	 net,	net,   0, arg)	\
+	/** I/O operations */			\
+	ACTION(DB_IO,	 io,	io,    0, arg)	\
+	/** Test debug stream */		\
+	ACTION(DB_TEST,	 test,  test,  0, arg)
+
+/** These macros are used intended to be used with FOREACH macros that define
+ *  log facilities in your library.   It will utilize internal FOREACH macros
+ *  to define debug bits as well as user defined macros.  See
+ *  D_FOREACH_GURT_FAC for an example.
+ *
+ * These macros can also be used standalone but are not quite as convenient
+ * due to required but unused arguments
+ */
+
+#ifdef D_LOG_USE_V2 /* Macros for D_DEBUG version 2 */
+/** Internal macro to declare a facility cache.  Used by DD_DECLARE_FAC */
+#define _DECLARE_CACHE(s_name, user_dbg_bits)				\
+	enum {								\
+		DD_FLAG_NAME(s_name, START),				\
+		D_FOREACH_PRIO_MASK(DECLARE_ENUM, DD_FAC(s_name))	\
+		D_FOREACH_DB(DECLARE_ENUM, DD_FAC(s_name))		\
+		user_dbg_bits(DECLARE_ENUM, DD_FAC(s_name))		\
+		DD_FLAG_NAME(s_name, END),				\
+	};								\
+	extern int DD_CACHE(DD_FAC(s_name))[DD_FLAG_NAME(s_name, END) - 1];
+
+/** Internal macro to create a facility cache.  Used by DD_INSTANTIATE_FAC */
+#define _INSTANTIATE_CACHE(s_name, user_dbg_bits)			\
+	int DD_CACHE(DD_FAC(s_name))[DD_FLAG_NAME(s_name, END) - 1] = {	\
+		D_FOREACH_PRIO_MASK(_INITIALIZE_FIELD, D_NOOP)		\
+		D_FOREACH_DB(_INITIALIZE_FIELD, D_NOOP)			\
+		user_dbg_bits(_INITIALIZE_FIELD, D_NOOP)		\
+	};
+
+/** Internal macro to register a facility cache.  Used by
+ *  ALLOCATE_LOG_FACILITY
+ */
+#define _ADD_CACHE(s_name)					\
+	d_log_add_cache(DD_CACHE(DD_FAC(s_name)),		\
+			ARRAY_SIZE(DD_CACHE(DD_FAC(s_name))));
+#else /* !D_LOG_USE_V2 */
+#define _ADD_CACHE(s_name) (void)0
+#define _INSTANTIATE_CACHE(s_name, user_dbg_bits) 
+#define _DECLARE_CACHE(s_name, user_dbg_bits) 
+#endif /* D_LOG_USE_V2 */
+
+/** Declare an extern to global facility variable
+ *
+ * \param s_name	Short name for the facility
+ * \param l_name	Unused
+ * \param user_dbg_bits	Unused
+ */
+#define DD_DECLARE_FAC(s_name, l_name, user_dbg_bits)	\
+	extern int DD_FAC_DECL(s_name);			\
+	_DECLARE_CACHE(s_name, user_dbg_bits)
+
+/** Declare a global facility variable
+ *
+ * \param s_name	Short name for the facility
+ * \param l_name	Unused
+ * \param user_dbg_bits	Unused
+ */
+#define DD_INSTANTIATE_FAC(s_name, l_name, user_dbg_bits)	\
+	int DD_FAC_DECL(s_name);				\
+	_INSTANTIATE_CACHE(s_name, user_dbg_bits)
+
+
+/** Internal use macro as part of DD_REGISTER_FAC.  It's multiline on purpose
+ *  so we can break from the foreach loop.  Checkpatch will complain but don't
+ *  see a better way.
+ */
+#define ALLOCATE_LOG_FACILITY(s_name, l_name, rc)			\
```

Comment:
(style)  Macros with multiple statements should be enclosed in a do - while loop

(style)  macros should not use a trailing semicolon